src/date-range-picker/calendar/grids/index.tsx

```diff
@@ -85,39 +86,27 @@ export const Grids = ({
     }
   }, [baseDate, focusedDate, isSingleGrid, isDateEnabled, onFocusedDateChange]);
 
-  const onGridKeyDownHandler = (e: React.KeyboardEvent) => {
+  const onGridKeyDownHandler = (event: React.KeyboardEvent) => {
     let updatedFocusDate;
 
-    if (focusedDate === null) {
+    const keys = [KeyCode.up, KeyCode.down, KeyCode.left, KeyCode.right, KeyCode.space, KeyCode.enter];
+
+    if (focusedDate === null || keys.indexOf(event.keyCode) === -1) {
       return;
     }
 
-    switch (e.keyCode) {
-      case KeyCode.space:
-      case KeyCode.enter:
-        e.preventDefault();
-        if (focusedDate) {
-          onSelectDate(focusedDate);
-        }
-        return;
-      case KeyCode.right:
-        e.preventDefault();
-        updatedFocusDate = moveNextDay(focusedDate, isDateEnabled);
-        break;
-      case KeyCode.left:
-        e.preventDefault();
-        updatedFocusDate = movePrevDay(focusedDate, isDateEnabled);
-        break;
-      case KeyCode.up:
-        e.preventDefault();
-        updatedFocusDate = movePrevWeek(focusedDate, isDateEnabled);
-        break;
-      case KeyCode.down:
-        e.preventDefault();
-        updatedFocusDate = moveNextWeek(focusedDate, isDateEnabled);
-        break;
-      default:
-        return;
+    event.preventDefault();
+
+    handleKeyDown({
+      onActivate: () => focusedDate && onSelectDate(focusedDate),
+      onBlockEnd: () => focusedDate && (updatedFocusDate = moveNextWeek(focusedDate, isDateEnabled)),
+      onBlockStart: () => focusedDate && (updatedFocusDate = movePrevWeek(focusedDate, isDateEnabled)),
+      onInlineEnd: () => focusedDate && (updatedFocusDate = moveNextDay(focusedDate, isDateEnabled)),
+      onInlineStart: () => focusedDate && (updatedFocusDate = movePrevDay(focusedDate, isDateEnabled)),
+    })(event);
```

Comment:
Not sure about this `)(event)` syntax. Usually people do not read the trailing part and this is where bugs can hide.

This is more error proof

```js
handleKey(error, {
  onActivate,
   ...etc
});
```

Also the code inside has nothing to do with `down` event, so let's call it just `key` so we can reuse it in keyup and keypress

Reply:
I updated the function to `handleKey` across all instances. The `(event`) syntax is effectively calling the `handleKey` function on the same line as the declaration. The reason it is not called directly is because it is nested inside a keydown handler that is trigged by the `onKeyDown` of the component which contains additional logic. I tried the above but that doesn't actually call the `handleKey` with the event. Is there a more appropriate syntax?

Reply:
```
handleKey({ ... })(event);
let myHandleKey = handleKey({ ... });
myHandleKey(event);
```
To my knowledge, these are the same?